### PR TITLE
Added a checkbox under Package Manager Settings to allow users to see debug logs

### PR DESCRIPTION
**Pull request by jainaashish**
Added a checkbox under Package Manager Settings to allow users to see debug logs.

Fix https://github.com/NuGet/Home/issues/2705

@joelverhagen @emgarten @alpaix @harikm86 @rrelyea

### CODE REVIEW

**Comment by emgarten** [comment]
@jainaashish can you share a screenshot of this change?

It looks like this writes the setting out to nuget.config, this would mean that nuget.exe and xplat would need to follow it also, which could conflict with the passed in value. It would be easier to persist this in the VS solution per user settings, NuGet stores a few other things for the UI in there already.

**Comment by harikmenon** [comment]
Actually I think this should be a VS side setting, just like MSBuild. They can turn it off and on. It definitely should not persist in the nuget.config or even in the solution IMO.

**Comment by emgarten** [comment]
Discussed offline, this shouldn't be in the nuget.config file since that would apply to the entire repository. Ideally this should be a per user VS wide setting. This could go into the suo file, but that would be per solution. Maybe there is another place to store global VS per user settings?

**Comment by jainaashish** [comment]
Suo file might not work here since it will then become per solution. There're global VS per user setting, let me check for that. And I'll use that instead of nuget.config.

**Comment by jainaashish** [comment]
After discussing with Yishai and Hari, we decided to use msbuild output verbosity level to control logging for nuget package manager ui actions.

File: src/NuGet.Clients/VsExtension/OutputConsoleLogger.cs [hunk 1]
```diff
@@ -21,8 +22,20 @@ internal class OutputConsoleLogger : INuGetUILogger
...
+
+        private string _vsVersion;
+
         private const string LogEntrySource = "NuGet Package Manager";
 
+        private const string VisualStudioRegKey = @"Software\Microsoft\VisualStudio";
+
+        private const string GeneralRegKey = "General";
+
+        private const string MSBuildVerbosityKey = "MSBuildLoggerVerbosity";
```
Comment (emgarten) [inline src/NuGet.Clients/VsExtension/OutputConsoleLogger.cs]: This can come from DTE see: GetMSBuildOutputVerbositySetting

**Comment by emgarten** [comment]
Change this to use DTE, then :shipit:

**Comment by jainaashish** [comment]
ffef3c3893f461ea311ed4803a464190d2bb4518

**Pull request closed without merging**